giskard_vision/detectors/base.py

```diff
@@ -0,0 +1,145 @@
+from abc import abstractmethod
+from dataclasses import dataclass
+from typing import Any, List, Optional, Sequence, Tuple
+
+try:
+    from giskard.scanner.issues import Issue, IssueGroup, IssueLevel
+    from giskard.scanner.registry import Detector
+except (ImportError, ModuleNotFoundError) as e:
+    e.msg = "Please install giskard to use custom detectors"
+    raise e
```

Comment:
This will break the installation (see https://github.com/Giskard-AI/giskard-vision/actions/runs/7584923434/job/20659661080)

These have to be imported internally (inside the class even if many times) or in worst case scenario: `except: pass`